### DISCUSSION TITLE
gammu-smsd -> after sending SMS "/ttyUSB<x>" changes

### DISCUSSION THREAD

**Discussion by CrazyVaclav** [gammu/gammu#706] [Q&A]
- OS: Raspbian GNU/Linux 10 (buster)
- Gammu-smsd version 1.40.0 installed with apt-get install
- UMTS-Stick Huawei E173

Hi,
every time I successfully send the first SMS with `echo "test"  | sudo gammu-smsd-inject TEXT number` the 
`/dev/serial/by-id/usb-HUAWEI_Technology_HUAWEI_Mobile-if00-port0` toggles between `ttyUSB0` and `ttyUSB1`

That means if I want to send a second SMS I had to change the `port = ` parameter

- I tried to set `port = /dev/serial/by-id/usb-HUAWEI_Technology_HUAWEI_Mobile-if00-port0` but I always have to restart the gammu-smsd.service after sending an SMS that the new correct ttyUSB is sourced.

**Can anyone help me to disable the toogle the ttyUSB or how gammu-smsd can auto reload it after every SMS** ?

Here is my gammu-smsdrc config File:

```
# Gammu library configuration, see gammurc(5)
[gammu]
# Please configure this!
port = /dev/serial/by-id/usb-HUAWEI_Technology_HUAWEI_Mobile-if00-port0
connection = at
# Debugging
logformat = textalldate
# SMSD configuration, see gammu-smsdrc(5)
[smsd]
pin = xxxx
NetworkCode = xxxxxxxxx
service = files
#logfile = syslog
logfile = /var/log/gammu-smsd
# Increase for debugging information
debuglevel = 4
ReceiveFrequency = 300
# Paths where messages are stored
inboxpath = /var/spool/gammu/inbox/
outboxpath = /var/spool/gammu/outbox/
sentsmspath = /var/spool/gammu/sent/
errorsmspath = /var/spool/gammu/error/
CheckSecurity = 0
```

**Comment by gomanuke** [comment]
I've got a similar problem: ZTE modem switches occasionally (not every SMS send). I've noticed, that **use_locking = no** changes something...

**Reply by CrazyVaclav** [reply]
As I only want to send SMS form the Command Line and do not need to receive SMS and or store them in a Database I switched to the "gammu" package. I removed  the package "gammu-smsd" and followed this [instructions](https://escapologybb.com/send-sms-from-raspberry-pi/)

I send multible test SMS and did a reboot so far there is no switching so for me the Problem just solved itself by using another package
